File: app/src/main/java/com/example/kineduexample/data/network/KineduInteractor.java
package com.example.kineduexample.data.network;

import com.example.kineduexample.data.network.model.ActivitiesIndex;
import com.example.kineduexample.data.network.model.ArticleDetailsIndex;
import com.example.kineduexample.data.network.model.ArticlesIndex;
import retrofit2.Call;

public interface KineduInteractor {
    Call<ActivitiesIndex> getActivities();
    Call<ArticlesIndex> getArticles();
    Call<ArticleDetailsIndex> getArticleDetails(Integer articleId);
}

File: app/src/main/java/com/example/kineduexample/data/network/KineduInteractor.kt
package com.example.kineduexample.data.network

import com.example.kineduexample.data.network.model.ActivitiesIndex
import com.example.kineduexample.data.network.model.ArticleDetailsIndex
import com.example.kineduexample.data.network.model.ArticlesIndex
import retrofit2.Call

interface KineduInteractor {
    fun getActivities(): Call<ActivitiesIndex>
    fun getArticles(): Call<ArticlesIndex>
    fun getArticleDetails(articleId: Int?): Call<ArticleDetailsIndex>
}
